mitmproxy/net/http/http1/assemble.py

```diff
@@ -45,31 +45,26 @@ def _assemble_request_line(request_data):
     Args:
         request_data (mitmproxy.net.http.request.RequestData)
     """
-    form = request_data.first_line_format
-    if form == "relative":
+    if request_data.method.upper() == b"CONNECT":
         return b"%s %s %s" % (
             request_data.method,
-            request_data.path,
+            request_data.authority,
             request_data.http_version
         )
-    elif form == "authority":
-        return b"%s %s:%d %s" % (
+    elif request_data.authority:
+        return b"%s %s://%s%s %s" % (
             request_data.method,
-            request_data.host,
-            request_data.port,
+            request_data.scheme,
+            request_data.authority,
+            request_data.path,
             request_data.http_version
         )
-    elif form == "absolute":
-        return b"%s %s://%s:%d%s %s" % (
+    else:
+        return b"%s %s %s" % (
             request_data.method,
-            request_data.scheme,
-            request_data.host,
-            request_data.port,
             request_data.path,
             request_data.http_version
```

Comment:
Any chance we can use fancy new Python f-strings?

Reply:
Unfortunately no, they do not work for bytestrings.